bigtable-hbase-parent/bigtable-hbase/src/main/java/com/google/cloud/bigtable/hbase/adapters/PutAdapter.java

```diff
@@ -57,56 +57,55 @@ public PutAdapter(int maxKeyValueSize, boolean setClientTimestamp) {
       throw new IllegalArgumentException("No columns to insert");
     }
 
-    MutateRowRequest.Builder result = MutateRowRequest.newBuilder();
-    result.setRowKey(ByteString.copyFrom(operation.getRow()));
-
     // Bigtable uses a 1ms granularity. Use this timestamp if the Put does not have one specified to
     // make mutations idempotent.
     long currentTimestampMicros = setClientTimestamp ? System.currentTimeMillis() * 1000 : -1;
+    final int rowLength = operation.getRow().length;
 
+    List<Mutation> mutations = new ArrayList<>();
     for (Entry<byte[], List<Cell>> entry : operation.getFamilyCellMap().entrySet()) {
       ByteString familyString = ByteString.copyFrom(entry.getKey());
+      int familySize = familyString.size();
 
       for (Cell cell : entry.getValue()) {
+        int qualifierLength = cell.getQualifierLength();
+        int valueLength = cell.getValueLength();
         // Since we are not using the interface involving KeyValues, we reconstruct how big they would be.
         // 20 bytes for metadata plus the length of all the elements.
-        int keyValueSize = (20 +
-            cell.getRowLength() +
-            cell.getFamilyLength() +
-            cell.getQualifierLength() +
-            cell.getValueLength());
+        int keyValueSize = (20 + rowLength + familySize + qualifierLength + valueLength);
```

Comment:
Where does the magic 20 come from? Could we hoist it into `BigtableConstants`?

Also, would it make sense to put this calculation into a private static getKeyValueSize method that takes an operation and cell?